webui/react/src/stores/userSettings.tsx

```diff
@@ -141,8 +148,40 @@ class UserSettingsStore extends PollingStore {
     }
   }
 
+  /**
+   * This updates the value of a setting and persists it for future sessions.
+   * @param type The type of the value or an encoder of the value to JSON.
+   * @param key Unique key to store and retrieve the settings
+   * @param fn Function to update the value of the setting at `key`
+   */
+  public update<T>(type: t.Type<T, Json>, key: string, fn: (value: T | undefined) => T): void {
+    this.#settings.update((settings) => {
+      return Loadable.map(settings, (settings) => {
+        return settings.update(key, (jsonValue) => {
+          let value: T | undefined = undefined;
+          if (jsonValue !== undefined) {
+            const attempt = type.decode(jsonValue);
+            // Silently discard incorrectly formatted values
+            if (isRight(attempt)) {
+              value = attempt.right;
+            } else {
+              console.error(`Setting at key '${key}' could not be decoded as ${type.name}`);
+            }
+          }
+          const newValue = fn(value);
+          // This is non-blocking. If the API call fails we don't want to block
+          // the user from interacting, just let them know that their settings
+          // are not persisting. It's also important to update the value immediately
+          // for good rendering performance.
+          setTimeout(() => this.updateUserSetting(key, newValue), 0);
+          return type.encode(newValue);
+        });
+      });
+    });
+  }
+
   /** Clears the setting, returning it to `null`. */
-  public remove(key: string) {
+  public remove(key: string): void {
     this.updateUserSetting(key, null);
```

Comment:
If `setTimeout` is needed for `set` and `update`, should it be applied here as well?